app/components/Task.tsx

```diff
@@ -139,24 +140,115 @@ const Task = () => {
         <div className='h-full space-y-3'>
           {list.length > 0 ? (
             list.map((item, index) => (
-              <div key={index} className='flex justify-center items-center'>
-                <div className='w-full  border-2'>
-                  <div>{item.title}</div>
-                  <div>
-                    {item.tasks.length > 0 ? (
-                      <div key={index} className='flex flex-col'>
-                        {item.tasks[0]} {/* IF HAVE TASK */}
-                      </div>
+              <div
+                key={index}
+                className='flex justify-center items-center p-3 rounded-md bg-neutral-300'
+              >
+                <Dialog>
+                  <DialogTrigger className='w-full text-left'>
+                    <h3>{item.title}</h3>
+                    <div>
+                      {item.tasks.length > 0 ? (
+                        <div key={index} className='flex flex-col'>
+                          {item.tasks[0]} {/* IF HAVE TASK */}
+                        </div>
+                      ) : (
+                        'no task' // IF NO TASK
+                      )}
+                    </div>
+                  </DialogTrigger>
+                  <DialogContent>
+                    <h2>{item.title}</h2>
+                    {item.tasks[0].length > 0 ? ( // need fix
+                      // LIST OF TASK W/ CHECKBOX
+                      item.tasks.map((task, index) => (
+                        <div key={index} className='flex justify-between'>
+                          <p>{task}</p>
+                          {/* when tick, crossout text */}
+                          <input type='checkbox' name={`task`} />
+                        </div>
+                      ))
                     ) : (
-                      'null' // IF NO TASK
+                      <p>no task</p> // IF NO TASK
                     )}
-                  </div>
-                </div>
+                    <DialogClose>close</DialogClose>
+                  </DialogContent>
+                </Dialog>
                 {/* EDIT AND DELETE BTNS */}
                 <div className='flex space-x-3 items-center'>
-                  <button className='bg-green-500 text-white p-1 rounded-sm size-8'>
-                    <Edit />
-                  </button>
+                  {/* EDIT LIST */}
+                  <Dialog>
+                    <DialogTrigger className='bg-green-500 text-white p-1 rounded-sm size-8'>
+                      <Edit />
+                    </DialogTrigger>
+                    <DialogContent>
+                      <div className='w-full space-y-3'>
+                        <h2 className='flex items-center justify-center'>
+                          Edit
+                        </h2>
+                        {/* FORM TITLE */}
```

Comment:
instead of adding comments, try to refactor the code and make it into a component

for example:
`<div className='flex  space-x-3 items-center'>
  <p>Title:</p>
  <input
    type='text'
    value={item.title}
    // onChange={(e) => setUserTitleInput(e.target.value)}
    placeholder='Add title here'
    className='border-2 p-1'
  />
</div>`

this section can be moved to app/components/FormTitle

Reply:
do it for your other {/* comments */} as well